project/SparkBuild.scala

```diff
@@ -311,6 +311,7 @@ object SparkBuild extends PomBuild {
 
     (Compile / javacOptions) ++= Seq(
       "-encoding", UTF_8.name(),
+      "-g",
```

Comment:
This looks safe to me. BTW, we still have differences, don't we?
> SBT does not have this behavior by default, which leads to some differences between the jars of maven and sbt builds, 

The file size seems to be increased from 386053 to 387076 with this, but it's still smaller than Maven-generated size, `397996`.

Reply:
BTW, this is bigger than the PR title.
> Preserve parameter information when using SBT package jar

1. If this is only for parameter info, `-parameters` might be better.
2. However, to be consistent Maven, `-g` is correct. 

Given (2), I guess we need to revise the PR title, @cxzl25 .

Reply:
>  we need to revise the PR title

OK

> but it's still smaller than Maven-generated size

In this PR, the main difference between the two is the META-INF folder, maven generated more information, but SBT did not.


<img width="251" alt="image" src="https://github.com/apache/spark/assets/3898450/8e2466ac-9962-4c96-a4e5-70cf59bf9dc3">


<img width="1017" alt="image" src="https://github.com/apache/spark/assets/3898450/28d038cd-5949-485a-bf9d-0c8b1edbfdfe">